# MOMENTUM 여름방학 정규세션 5주차

타이타닉호 생존자 예측을 통해 알아보는 머신러닝 분류 모델.

## 데이터 셋 준비하기

[`kaggle-api`](https://github.com/Kaggle/kaggle-api)를 이용해서 간단하게 받을 수 있습니다.

`titanic/` 폴더를 만든 뒤, 타이타닉 데이터 셋을 내려받아서 이 폴더에 저장해보겠습니다.

In [4]:
!pip install kaggle
!kaggle competitions download -c titanic

import zipfile
import os

file_name = "titanic.zip"

# 타이타닉 데이터셋 .zip 파일이 있는지 검사
assert os.path.exists(file_name)

zip_file = zipfile.ZipFile(file_name)
zip_file.extractall(path='./titanic/')

# 압축풀기 완료 후에는 .zip 파일 제거.
os.remove(file_name)

You should consider upgrading via the '/Users/hepheir/GitHub/smu-momentum/practice-ml-titanic/.venv/bin/python3.10 -m pip install --upgrade pip' command.
titanic.zip: Skipping, found more recently modified local copy (use --force to force download)


## Pandas로 타이타닉 데이터 불러오기

데이터가 잘 불러져왔는지 확인해볼까요?

In [11]:
# 타이타닉 데이터셋 불러오기

import pandas as pd

train = pd.read_csv('./titanic/train.csv')
test = pd.read_csv('./titanic/test.csv')

print('================================================================')
print('Train 데이터 셋')
print('================================================================')
train.info(memory_usage='deep')


print('================================================================')
print('Test 데이터 셋')
print('================================================================')
test.info(memory_usage='deep')

Train 데이터 셋
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 315.0 KB
Test 데이터 셋
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 418 entries, 0 to 417
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  418 non-null  

타이타닉 데이터 셋에는 어떤 열과 행이 있는지 살펴봅시다.

* 훈련용 데이터 셋(`train.csv`)에는 12개의 열과 891개의 행이 존재하고,
* 검증용 데이터 셋(`test.csv`)에는 11개의 열과 419개의 행이 존재하는군요.

## 데이터를 받으면 가장 먼저 할 일

빠르게 데이터를 살펴보면서 결측치나 이상치가 존재하는지 알아봅시다.

In [14]:
# 학습데이터의 결측치 확인

# 결측 여부에 대한 True/False 가 표기된 데이터 프레임을 만든다.
df = train.isnull()

# Boolean(참/거짓) 값은 산술연산으로 처리될 때 True = 1, False = 0 로서 처리되는 성질을 이용하여 True(결측치이다)의 개수를 센다.
series = df.sum()

# Series 타입을 DataFrame으로 변환. `name=` 에 넣는 이름이 Series가 갖고 있던 값의 칼럼 이름이 된다.
df = series.reset_index(name='결측치')

# 보기 좋게 출력해보자
df

index  결측치
0   PassengerId    0
1      Survived    0
2        Pclass    0
3          Name    0
4           Sex    0
5           Age  177
6         SibSp    0
7         Parch    0
8        Ticket    0
9          Fare    0
10        Cabin  687
11     Embarked    2